str.go

```diff
@@ -0,0 +1,85 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+var (
+	ErrorBadEpoch = errors.New("[merkletree] Bad STR Epoch. STR's epoch must be nonzero")
+)
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous epochs, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	tree        *MerkleTree
+	epoch       uint64
+	prevEpoch   uint64
+	prevStrHash []byte
+	sig         []byte
+	policies    Policies
+}
+
+func NewSTR(key crypto.SigningKey, policies Policies, m *MerkleTree, epoch uint64, prevHash []byte) *SignedTreeRoot {
+	if epoch < 1 {
+		panic(ErrorBadEpoch)
+	}
+	str := &SignedTreeRoot{
+		tree:        m,
+		epoch:       epoch,
+		prevEpoch:   epoch - 1,
+		prevStrHash: prevHash,
+		policies:    policies,
+	}
+	bytesPreSig := str.getSTRBytesForSig()
+	str.sig = crypto.Sign(key, bytesPreSig)
+	return str
+}
+
+func (str *SignedTreeRoot) getSTRBytesForSig() []byte {
+	var strBytes []byte
+	strBytes = append(strBytes, util.ULongToBytes(str.epoch)...)     // t - epoch number
+	strBytes = append(strBytes, util.ULongToBytes(str.prevEpoch)...) // t_prev - previous epoch number
+	strBytes = append(strBytes, str.tree.hash...)                    // root
+	strBytes = append(strBytes, str.prevStrHash...)                  // previous STR hash
+	strBytes = append(strBytes, str.policies.Serialize()...)         // P
+	return strBytes
+}
+
+// serialize encodes the str to a byte array with the following format:
+// [tree hash, epoch, previous epoch, previous hash, signature]
+func (str *SignedTreeRoot) serialize() []byte {
```

Comment:
Why is this defined this way?  Are policies purposely omitted?  I don't think this function is necessary, and if it is, it should be defined in terms of `getSTRBytesForSig+str.sig`.